### ISSUE TITLE
Bug: Paragraphs beginning with a html-comment do not trigger `two-spaces-between-lines-with-content`

### ISSUE THREAD

**Issue by Gewerd-Strauss** [platers/obsidian-linter#1238]
- [x] I have verified that I am on the latest version of the Linter

## Describe the Bug

Given the following text:

```md
<!-- Camera-Rotation/ -->
<!-- Introduction to different setups -->For experiments examining the influence of rotational offsets of the camera relative to the \acr{CPL}/\acr{IDP}/\acr{IDD}, @tbl-gt-table-table_camera_tilts_GFA-v4 state the modified parameters and their experimental groups.
The declared distances are relative to the center of the \acr{CPL}/\acr{IDP}/\acr{IDD}/camera-lens, and to the bottom-left corner of the photo-box. Where applicable, the orientation of a non-symmetrical object is listed. The dimensions of the photo-box are listed in @tbl-report-methods-general-baseline-setup-parameters.
```

No linebreak-double-space will be inserted at the end of the first sentence (that is, the line ends in ` the modified parameters and their experimental groups.`, not in ` the modified parameters and their experimental groups.  `).

This results in the paragraphs collapsing into each other when viewed/read/processed by a markdown processor.

If the html-comment at the beginning of the first line is removed:

```md
<!-- Camera-Rotation/ -->
For experiments examining the influence of rotational offsets of the camera relative to the \acr{CPL}/\acr{IDP}/\acr{IDD}, @tbl-gt-table-table_camera_tilts_GFA-v4 state the modified parameters and their experimental groups.
The declared distances are relative to the center of the \acr{CPL}/\acr{IDP}/\acr{IDD}/camera-lens, and to the bottom-left corner of the photo-box. Where applicable, the orientation of a non-symmetrical object is listed. The dimensions of the photo-box are listed in @tbl-report-methods-general-baseline-setup-parameters.
```
and the file is linted, we obtain

```md
<!-- Camera-Rotation/ -->
For experiments examining the influence of rotational offsets of the camera relative to the \acr{CPL}/\acr{IDP}/\acr{IDD}, @tbl-gt-table-table_camera_tilts_GFA-v4 state the modified parameters and their experimental groups.  
The declared distances are relative to the center of the \acr{CPL}/\acr{IDP}/\acr{IDD}/camera-lens, and to the bottom-left corner of the photo-box. Where applicable, the orientation of a non-symmetrical object is listed. The dimensions of the photo-box are listed in @tbl-report-methods-general-baseline-setup-parameters.
```

That is, the first paragraph ends in `the modified parameters and their experimental groups.  `.

Removing the linebreak-spaces, and re-adding the html-comment at the beginning of the line, then re-linting then reverts to scenario 1, where the double-spaces are not added.
